packages/cli/src/utils/getUrlListFromArgs.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies.
+ */
+import { readFile } from 'fs-extra';
+import { parseStringPromise } from 'xml2js';
+/**
+ * Internal dependencies.
+ */
+import Utility from './utility';
+
+/**
+ * Validate arguments passed to the CLI. Process for the CLI is exited with appropriate message.
+ * @param {string} url Url input to CLI.
+ * @param {string} sitemapUrl Url of a sitemap.
+ * @param {string} csvPath File system path to a csv file with urls.
+ * @param {string} sitemapPath File system path to a sitemap xml file.
+ * @param {string} numberOfUrls Url limit argument.
+ * @param {string} outDir File system path to the output directory.
+ */
+
+const getUrlListFromArgs = async (
+  url: string,
+  sitemapUrl: string,
+  csvPath: string,
+  sitemapPath: string,
+  // @ts-ignore Package does not support typescript.
+  spinnies
+): Promise<string[]> => {
+  let urls: string[] = [];
+
+  if (url) {
```

Comment:
Can we break the processing of each arg into smaller functions inside the same file?